blog/2024/storing-sessions-in-kc26.adoc

```diff
@@ -0,0 +1,204 @@
+:title: Storing sessions in Keycloak 26
+:date: 2024-12-03
+:publish: true
+:author: Michal Hajas
+
+As you may know, Keycloak 26 now uses by default the Persistent user sessions feature.
+In this blog post I would like to uncover a little bit more background on why we introduced this feature, what are the alternatives and what is the future.
+
+== Session storages in Keycloak 26 cheatsheet
+
+This section provides a TLDR guidance on what sessions storages exist and when each of them should be used with Keycloak 26.
+The following sections provide more details on each storage type and reasoning behind introducing or dropping each of them.
+
+++++
+<div class="kc-asciidoc">
+++++
+
+|===
+|Number of sites |Sessions storage |Characteristics |When to use |Keycloak CLI options to enable
+
+.4+.^| Single site
+|Persistent sessions
+a|
+* Sessions stored in the database and cached in memory
+* Sessions available after cluster restart
+* Lower memory usage
+* Higher database usage
+a|
+* Default and recommended for standard installations
+* You want your sessions to survive restarts and upgrades
+* Accept higher database usage
+|No additional configuration needed
+
+|Sessions stored in memory
+a|
+* Faster reads and writes
+* Sessions lost after cluster restart
+* Higher memory usage (all sessions must be in memory)
+a|
+* Can't use persistent sessions feature
+* Please provide your feedback https://github.com/keycloak/keycloak/discussions/28271[here], as we want to understand why you can't use persistent sessions
+a|
+----
+--features-disabled="persistent-user-sessions"
+----
+
+|Sessions stored in external Infinispan
+a|
+* Sessions stored only in external Infinispan
+* Reduced database usage
+* Using Hot Rod client for communication with external Infinispan
+* Experimental feature
+a|
+* Do not use in production as it is experimental
+* Evaluate and provide your feedback https://github.com/keycloak/keycloak/discussions/33745[here] if you are interested in this feature and want to help to make it supported.
+
+a|
+----
+--features="clusterless"
+--features-disabled="persistent-user-sessions"
+----
+
+|Sessions stored in memory and external Infinispan
+a|
+* 4 copies of each session 2x in Keycloak memory and 2x in Infinispan memory
+* Sessions available after Keycloak cluster restarts
+* High memory usage
+* Experimental and will be removed soon
+a|
+* When you used this setup with previous releases and cannot switch to persistent user sessions now
+a|
+----
+--features="cache-embedded-remote-store"
+--features-disabled="persistent-user-sessions"
+----
+.2+.^|Multiple sites (https://www.keycloak.org/high-availability/introduction[guide])
+|Persistent user sessions
```

Comment:
consistent naming: On line 22, it says `Persistent sessions`. My suggestion is to use the same name everywhere (no preference between the two).

Reply:
Thanks @pruivo, I changed all occurences of `persistent sessions` -> `persistent user sessions` to be on par with the name of the feature. Thanks!